From 6105b1aa50338d1380f7ccad4f6f007c77f6519f Mon Sep 17 00:00:00 2001
From: Jason Madden <jamadden@gmail.com>
Date: Fri, 13 Sep 2024 04:55:06 -0500
Subject: [PATCH] Remove unnecessary logging during interpreter shutdown. Fixes
 #426.

---
 src/greenlet/TThreadStateDestroy.cpp | 4 ++++
 1 files changed, 4 insertions(+)

diff --git a/src/greenlet/TThreadStateDestroy.cpp b/src/greenlet/TThreadStateDestroy.cpp
index 3aab2e7..e5a9661 100644
--- a/src/greenlet/TThreadStateDestroy.cpp
+++ b/src/greenlet/TThreadStateDestroy.cpp
@@ -108,9 +108,13 @@ struct ThreadState_DestroyNoGIL
 #else
         if (_Py_IsFinalizing()) {
 #endif
+#ifdef GREENLET_DEBUG
+            // No need to log in the general case. Yes, we'll leak,
+            // but we're shutting down so it should be ok.
             fprintf(stderr,
                     "greenlet: WARNING: Interpreter is finalizing. Ignoring "
                     "call to Py_AddPendingCall; \n");
+#endif
             return 0;
         }
         return Py_AddPendingCall(func, arg);
